File: practice/lesson2/Rectangle.java
package practice.lesson2;

public class Rectangle extends Figure implements Perimeter {
    private double sideA;
    private double sideB;

    public Rectangle(double sideA, double sideB) {
        this.sideA = sideA;
        this.sideB = sideB;
    }

    public double getSideA() {
        return this.sideA;
    }

    public double getSideB() {
        return this.sideB;
    }

    @Override
    public double getArea() {
        return sideA * sideB;
    }

    @Override
    public double getPerimeter() {
        return 2 * (sideA + sideB);
    }

    @Override
    public String toString() {
        String result = String.format("%s\nP: %.2f", super.getInfo(this), this.getPerimeter());
        return result;
    }

}

File: homeworks/lesson6/Rectangle.cs
public class Rectangle : Polygon
{

    public Rectangle(int a, int b) : base(new int[] {a, b, a, b})
    {
        
    }

    public Rectangle() : base(new int[] {3, 5, 3, 5})
    {
        
    }
    public override double area()
    {
        return base.sides[0] * this.sides[1];
    }

    public override string ToString()
    {
        return $"Прямоугольник\nДлины сторон: {base.sides[0]} {base.sides[1]}\nПериметр: {this.perimeter()}\nПлощадь: {this.area()}\n";
    }
}
